src/dmd/statementsem.d

```diff
@@ -3698,6 +3698,13 @@ package (dmd) extern (C++) final class StatementSemanticVisitor : Visitor
 
         if (oss.tok != TOK.onScopeExit)
         {
+            // https://issues.dlang.org/show_bug.cgi?id=23159
+            if (!global.params.useExceptions)
+            {
+                oss.error("cannot use `%s` with -betterC", Token.toChars(oss.tok));
```

Comment:
```suggestion
                oss.error("`%s` cannot be used with -betterC because exception handling requires druntime.", Token.toChars(oss.tok));
```
? Latter part can be dropped but we should stop writing error messages that start with "cannot "